### PR TITLE
FIR UAST: facade support and testing infra update

### PR DESCRIPTION
**Pull request by jsjeon**
In #4312, many legacy test data with top-level declarations were ignored (by adding a directive), since facade wasn't found. After #4345, FIR-based light class for facade is now searchable via LC utils. 1st commit converts light class for facade.

Since facade class is converted, we can track every legacy test input. 2nd commit enables all remaining legacy test datas (and thus mostly mechanical changes). This will help visualizing progress/changes whenever we add new stuff to FIR UAST.

3rd commit reverts https://github.com/JetBrains/kotlin/commit/291cd842b96fac8cfa8fb8e6bccfa5dbe4f110a3 since uses of that directive were gone.

To reduce the footprints of render/log files as well as to signal "identical" dumps from FIR UAST and FE1.0 UAST, 4th commit adds utils to clean up identical render/log files. (We already have some!)

### CODE REVIEW

File: plugins/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/declarations/FirKotlinUFile.kt [hunk 1]
```diff
@@ -34,12 +36,12 @@ class FirKotlinUFile(
...
-            // TODO: Script
-            // TODO: Facade: getOrCreateFirLightFacade()
-            return sourcePsi.declarations.mapNotNull { (it as? KtClassOrObject)?.toUClass() }
-        }
+    override val classes: List<UClass> by lz {
+        val facadeOrScriptClass = if (sourcePsi.isScript()) sourcePsi.script?.toLightClass() else sourcePsi.findFacadeClass()
+        val facadeOrScriptUClass = facadeOrScriptClass?.toUClass()?.let { listOf(it) } ?: emptyList()
+        val classes = sourcePsi.declarations.mapNotNull { (it as? KtClassOrObject)?.toUClass() }
+        facadeOrScriptUClass + classes
+    }
```
Comment (jsjeon) [inline plugins/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/declarations/FirKotlinUFile.kt:44]: Now `UFile` is also in its final shape I guess. My next step would be to think of ways to simplify/unify `UFile` and `UImportStatement` at each plugin.

File: plugins/uast-kotlin-fir/testData/declaration/importOnDemand.log.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 UFile (package = declaration)
     UImportStatement (isOnDemand = true)
-    UImportStatement (isOnDemand = true)
+    UImportStatement (isOnDemand = true)
```
Comment (jsjeon) [inline plugins/uast-kotlin-fir/testData/declaration/importOnDemand.log.txt:3]: As expected(?), `UImportStatment` is in its final shape, and indeed dumps are identical. (Changes due to `trim()`.)

**Review by jsjeon — COMMENTED** [review]
cc @darthorimar @xiexed

**Comment by codeschre** [comment]
Join Zoom Meeting
https://us02web.zoom.us/j/3631983496?pwd=czNOcElQUGhMKzZVODEvdUVjZ0YwZz09

Meeting ID: 363 198 3496
Passcode: 1Pshi7

**Comment by darthorimar** [comment]
Thank you!

**Pull request merged**